MSAL/src/MSALPublicClientApplication.m

```diff
@@ -312,12 +318,16 @@ - (BOOL)setupTokenCacheWithConfiguration:(MSALPublicClientApplicationConfig *)co
         {
             MSID_LOG_WITH_CTX_PII(MSIDLogLevelWarning, nil, @"Failed to create secondary data source with error %@", MSID_PII_LOG_MASKABLE(secondaryDataSourceError));
         }
+        
+        self.msidCacheConfig = [[MSIDCacheConfig alloc] initWithKeychainGroup:config.cacheConfig.keychainSharingGroup accessRef:(__bridge SecAccessRef _Nullable)(secondaryDataSource.accessControlForNonSharedItems)];
```

Comment:
I don't think we need to fallback to secondary cache for this.
The worst thing that will happen is that we have to get a new access token after 10.14 -> 10.15 update.
IMO, performance hit we're introducing here is not worth it. So I'm fine regenerating keys and access tokens on 10.14 -> 10.15 update.

Reply:
Addressed. As discussed , added common code plus mac specific initializer.